src/groundlight/client.py

```diff
@@ -76,10 +71,19 @@ def __init__(self, endpoint: Optional[str] = None, api_token: Optional[str] = No
         self.detectors_api = DetectorsApi(self.api_client)
         self.image_queries_api = ImageQueriesApi(self.api_client)
 
-    def _fixup_image_query(self, iq: ImageQuery) -> ImageQuery:  # pylint: disable=no-self-use
-        """Process the wire-format image query to make it more usable."""
+    def _fixup_image_query(self, iq: ImageQuery, want_async: bool = False) -> ImageQuery:  # pylint: disable=no-self-use
+        """
+        Post process an image query prior to returning it to the user.
+        :param iq: The image query to fix up.
+        :param want_async: post processing for want_async=True option in submit_image_query
+        """
         # Note: This might go away once we clean up the mapping logic server-side.
         iq.result.label = convert_internal_label_to_display(iq, iq.result.label)
+        if want_async:
```

Comment:
Nice and clean.

Reply:
with the serverside change (where the serve returns result=None, this is no longer necessary, so it's even cleaner!